File: thru-hiker-log-app/src/app/services/entry.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { IEntry } from '../models/entry';

@Injectable({ providedIn: 'root' })
export class EntryService {
  private baseUrl = 'https://localhost:7225/api/entry';

  constructor(private http: HttpClient) { }

  /** Create an entry */
  createEntry(entry: IEntry): Observable<void> {
    return this.http.post<void>(this.baseUrl, entry);
  }

  /** Get entries for a specific trail */
  getEntriesByTrailId(trailId: number): Observable<IEntry[]> {
    return this.http.get<IEntry[]>(this.baseUrl + '/trail/' + trailId);
  }

  /** Get details for a specific entry */
  getEntryDetails(trailId: number, entryId: number): Observable<IEntry> {
    return this.http.get<IEntry>(`${this.baseUrl}/${entryId}/trail/${trailId}`);
  }

  /** Update an entry */
  updateEntry(entry: IEntry): Observable<void> {
    return this.http.put<void>(`${this.baseUrl}/${entry.id}/trail/${entry.trailId}`, entry);
  }

  /** Delete an entry */
  deleteEntry(trailId: number, entryId: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${entryId}/trail/${trailId}`);
  }
}

File: ThruHikerLogApi/ThruHikerLogApi/Services/EntryService.cs
using ThruHikerLogApi.Repos;
using ThruHikerLogApi.Models;

namespace ThruHikerLogApi.Services
{
    public class EntryService(EntryRepo entryRepo)
    {
        private readonly EntryRepo _entryRepo = entryRepo;

        /// <summary>
        /// Gets all entries for a specific trail
        /// </summary>
        /// <param name="trailId"></param>
        /// <returns></returns>
        public IEnumerable<Entry> GetEntriesForTrail(int trailId)
        {
            return _entryRepo.GetEntriesByTrailId(trailId);
        }

        /// <summary>
        /// Creates a new entry
        /// </summary>
        /// <param name="entry"></param>
        /// <returns></returns>
        public Entry CreateEntry(Entry entry)
        {
            return _entryRepo.CreateEntry(entry);
        }

        public Entry GetEntryDetails(int entryId, int trailId)
        {
            return _entryRepo.GetEntryDetails(entryId, trailId);
        }

        /// <summary>
        /// Updates an entry
        /// </summary>
        /// <param name="entry"></param>
        public void UpdateEntry(int entryId, int trailId, Entry entry)
        {
            _entryRepo.UpdateEntry(entryId, trailId, entry);
        }

        /// <summary>
        /// Updates an entry
        /// </summary>
        /// <param name="id"></param>
        public void DeleteEntry(int entryId, int trailId)
        {
            _entryRepo.DeleteEntry(entryId, trailId);
        }
    }
}
